nvme/src/lib.rs

```diff
@@ -0,0 +1,118 @@
+// NVMe completion status code type
```

Comment:
We want to expose an NVMe spec crate that is separate from libnvme itself since there are other possible consumers such as propolis.  In addition the identity.rs file's contents will also eventually be moved into the nvme crate.

The first consumer of some of these definitions can be seen here: https://github.com/oxidecomputer/nvme-firmware-updater/blob/f34d776167df3e598f3e9b0e4e7bdd30c8d34867/src/main.rs#L136-L155